File: Chat/ChatParser/src/SDKMessageDTO.cpp
/* 
 * File:   MessageDTO.h
 * Author: towhid
 *
 * Created on May 7, 2016, 2:35 PM
 */

#include "SDKMessageDTO.h"

namespace imsdk {
SDKMessageDTO::SDKMessageDTO() {
}

SDKMessageDTO::SDKMessageDTO(UserIdentity senderId, UserIdentity receiverId, GroupIdentity groupId, std::string packetId, int messageType, int timeOut, std::string message, long long messageDate, std::string memberFullName, std::string memberProfileUrl, bool isSecretVisible, int messageStatus, int direction, bool isEdited, int publicChatLikeCount, bool publicChatILike, bool publicChatIReport, int unreadCount) {
    this->senderId = senderId;
    this->receiverId = receiverId;
    this->groupId = groupId;
    this->packetId = packetId;
    this->messageType = messageType;
    this->timeOut = timeOut;
    this->message = message;
    this->messageDate = messageDate;
    this->memberFullName = memberFullName;
    this->memberProfileUrl = memberProfileUrl;
    this->isSecretVisible = isSecretVisible;
    this->messageStatus = messageStatus;
    this->direction = direction;
    this->isEdited = isEdited;
    this->publicChatLikeCount = publicChatLikeCount;
    this->publicChatILike = publicChatILike;
    this->publicChatIReport = publicChatIReport;
    this->unreadCount = unreadCount;
}

std::string SDKMessageDTO::getMemberFullName() const {
    return memberFullName;
}

void SDKMessageDTO::setMemberFullName(std::string memberFullName) {
    this->memberFullName = memberFullName;
}

std::string SDKMessageDTO::getMemberProfileUrl() const {
    return memberProfileUrl;
}

void SDKMessageDTO::setMemberProfileUrl(std::string memberProfileUrl) {
    this->memberProfileUrl = memberProfileUrl;
}

std::string SDKMessageDTO::getMessage() const {
    return message;
}

void SDKMessageDTO::setMessage(std::string message) {
    this->message = message;
}

long long SDKMessageDTO::getMessageDate() const {
    return messageDate;
}

void SDKMessageDTO::setMessageDate(long long messageDate) {
    this->messageDate = messageDate;
}

int SDKMessageDTO::getMessageType() const {
    return messageType;
}

void SDKMessageDTO::setMessageType(int messageType) {
    this->messageType = messageType;
}

std::string SDKMessageDTO::getPacketId() const {
    return packetId;
}

void SDKMessageDTO::setPacketId(std::string packetId) {
    this->packetId = packetId;
}

UserIdentity SDKMessageDTO::getSenderId() const{
    return senderId;
}
void SDKMessageDTO::setSenderId(UserIdentity senderId){
    this->senderId = senderId;
}

void SDKMessageDTO::setReceiverId(UserIdentity receiverId) {
    this->receiverId = receiverId;
}

UserIdentity SDKMessageDTO::getReceiverId() const {
    return receiverId;
}

void SDKMessageDTO::setDirection(int direction) {
    this->direction = direction;
}

int SDKMessageDTO::getDirection() const {
    return direction;
}

void SDKMessageDTO::setMessageStatus(int messageStatus) {
    this->messageStatus = messageStatus;
}

int SDKMessageDTO::getMessageStatus() const {
    return messageStatus;
}

void SDKMessageDTO::setSecretVisible(bool secretVisible) {
    isSecretVisible = secretVisible;
}

bool SDKMessageDTO::getIsSecretVisible() const {
    return isSecretVisible;
}

void SDKMessageDTO::setTimeOut(int timeOut) {
    this->timeOut = timeOut;
}

int SDKMessageDTO::getTimeOut() const {
    return timeOut;
}
void SDKMessageDTO::setIsEdited(bool isEdited){
    this->isEdited = isEdited;
}

bool SDKMessageDTO::getIsEdited() const{
    return isEdited;
}

void SDKMessageDTO::setPublicChatLikeCount(int publicChatLikeCount) {
    this->publicChatLikeCount = publicChatLikeCount;
}

int SDKMessageDTO::getPublicChatLikeCount() const {
    return publicChatLikeCount;
}

void SDKMessageDTO::setPublicChatILike(bool publicChatILike) {
    this->publicChatILike = publicChatILike;
}

bool SDKMessageDTO::isPublicChatILike() const {
    return publicChatILike;
}

void SDKMessageDTO::setPublicChatIReport(bool publicChatIReport) {
    this->publicChatIReport = publicChatIReport;
}

bool SDKMessageDTO::isPublicChatIReport() const {
    return publicChatIReport;
}

void SDKMessageDTO::setUnreadCount(int unreadCount) {
    this->unreadCount = unreadCount;
}

int SDKMessageDTO::getUnreadCount() const {
    return unreadCount;
}

void SDKMessageDTO::setGroupId(GroupIdentity groupId) {
    this->groupId = groupId;
}

GroupIdentity SDKMessageDTO::getGroupId() const {
    return groupId;
}
}

File: Chat/Chat/wrapper/java/JChatSDK/src/com/ringid/imsdk/SDKMessageDTO.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.ringid.imsdk;

/**
 *
 * @author towhid
 */
public class SDKMessageDTO {
    private long senderId;
    private long receiverId;
    private long groupId;
    private String packetId;
    private int messageType;
    private int timeOut;
    private String message;
    private long messageDate;
    private String memberFullName;
    private String memberProfileUrl;
    private boolean isSecretVisible;
    private int messageStatus;
    private int direction;
    private boolean isEdited;
    private int publicChatLikeCount;
    private boolean publicChatILike;
    private boolean publicChatIReport;
    private int unreadCount;

    public int getPublicChatLikeCount() {
        return publicChatLikeCount;
    }

    public void setPublicChatLikeCount(int publicChatLikeCount) {
        this.publicChatLikeCount = publicChatLikeCount;
    }

    public boolean isPublicChatILike() {
        return publicChatILike;
    }

    public void setPublicChatILike(boolean publicChatILike) {
        this.publicChatILike = publicChatILike;
    }

    public boolean isPublicChatIReport() {
        return publicChatIReport;
    }

    public void setPublicChatIReport(boolean publicChatIReport) {
        this.publicChatIReport = publicChatIReport;
    }
    
    public SDKMessageDTO(){
        
    }
    
    public SDKMessageDTO(long senderId, long receiverId, long groupId, String packetId, int messageType, int timeOut, String message, long messageDate, String memberFullName, String memberProfileUrl, boolean isSecretVisible, int messageStatus, int direction, boolean isEdited, int publicChatLikeCount, boolean publicChatILike, boolean publicChatIReport, int unreadCount) {
        this.senderId = senderId;
        this.receiverId = receiverId;
        this.groupId = groupId;
        this.packetId = packetId;
        this.messageType = messageType;
        this.timeOut = timeOut;
        this.message = message;
        this.messageDate = messageDate;
        this.memberFullName = memberFullName;
        this.memberProfileUrl = memberProfileUrl;
        this.isSecretVisible = isSecretVisible;
        this.messageStatus = messageStatus;
        this.direction = direction;
        this.isEdited = isEdited;
        this.publicChatLikeCount = publicChatLikeCount;
        this.publicChatILike = publicChatILike;
        this.publicChatIReport = publicChatIReport;
        this.unreadCount = unreadCount;           
    }

    public long getSenderId() {
        return senderId;
    }

    public void setSenderId(long senderId) {
        this.senderId = senderId;
    }

    public String getPacketId() {
        return packetId;
    }

    public void setPacketId(String packetId) {
        this.packetId = packetId;
    }

    public int getMessageType() {
        return messageType;
    }

    public void setMessageType(int messageType) {
        this.messageType = messageType;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public long getMessageDate() {
        return messageDate;
    }

    public void setMessageDate(long messageDate) {
        this.messageDate = messageDate;
    }

    public String getMemberFullName() {
        return memberFullName;
    }

    public void setMemberFullName(String memberFullName) {
        this.memberFullName = memberFullName;
    }

    public String getMemberProfileUrl() {
        return memberProfileUrl;
    }

    public void setMemberProfileUrl(String memberProfileUrl) {
        this.memberProfileUrl = memberProfileUrl;
    }

    /**
     * @return the receiverId
     */
    public long getReceiverId() {
        return receiverId;
    }

    /**
     * @param receiverId the receiverId to set
     */
    public void setReceiverId(long receiverId) {
        this.receiverId = receiverId;
    }

    /**
     * @return the timeOut
     */
    public int getTimeOut() {
        return timeOut;
    }

    /**
     * @param timeOut the timeOut to set
     */
    public void setTimeOut(int timeOut) {
        this.timeOut = timeOut;
    }

    /**
     * @return the isSecretVisible
     */
    public boolean isIsSecretVisible() {
        return isSecretVisible;
    }

    /**
     * @param isSecretVisible the isSecretVisible to set
     */
    public void setIsSecretVisible(boolean isSecretVisible) {
        this.isSecretVisible = isSecretVisible;
    }

    /**
     * @return the messageStatus
     */
    public int getMessageStatus() {
        return messageStatus;
    }

    /**
     * @param messageStatus the messageStatus to set
     */
    public void setMessageStatus(int messageStatus) {
        this.messageStatus = messageStatus;
    }

    /**
     * @return the direction
     */
    public int getDirection() {
        return direction;
    }

    /**
     * @param direction the direction to set
     */
    public void setDirection(int direction) {
        this.direction = direction;
    }
    
    public void setIsEdited(boolean isEdited){
        this.isEdited = isEdited;
    }
    
    public boolean getIsEdited(){
        return isEdited;
    }

    /**
     * @return the groupId
     */
    public long getGroupId() {
        return groupId;
    }

    /**
     * @param groupId the groupId to set
     */
    public void setGroupId(long groupId) {
        this.groupId = groupId;
    }

    /**
     * @return the unreadCount
     */
    public int getUnreadCount() {
        return unreadCount;
    }

    /**
     * @param unreadCount the unreadCount to set
     */
    public void setUnreadCount(int unreadCount) {
        this.unreadCount = unreadCount;
    }
    
}
